blivet/devices/storage.py

```diff
@@ -243,7 +243,7 @@ def _setTargetSize(self, newsize):
         if not isinstance(newsize, Size):
             raise ValueError("new size must of type Size")
 
-        if self.maxSize and newsize > self.maxSize:
+        if self.maxSize and newsize and newsize > self.maxSize:
```

Comment:
When does the factory set a device's size to 0? My recollection is that the factory's code for adjusting a container to deal w/ a removed device runs _after_ the device is removed.